src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -68,123 +67,161 @@ const LoadableVersionsSidebar = SidebarUtils.getAsyncSidebarContent(
     MARK_NAME_JS_LOADING_VERSIONS,
 );
 
-const SidebarPanels = ({
-    activitySidebarProps,
-    currentUser,
-    detailsSidebarProps,
-    file,
-    fileId,
-    getPreview,
-    getViewer,
-    hasActivity,
-    hasDetails,
-    hasMetadata,
-    hasSkills,
-    hasVersions,
-    isOpen,
-    metadataSidebarProps,
-    onVersionChange,
-    onVersionHistoryClick,
-    refreshIdentity,
-    versionsSidebarProps,
-}: Props) =>
-    isOpen && (
-        <Switch>
-            {hasSkills && (
-                <Route
-                    exact
-                    path={`/${SIDEBAR_VIEW_SKILLS}`}
-                    render={() => (
-                        <LoadableSkillsSidebar
-                            key={file.id}
-                            file={file}
-                            getPreview={getPreview}
-                            getViewer={getViewer}
-                            startMarkName={MARK_NAME_JS_LOADING_SKILLS}
+class SidebarPanels extends React.Component<Props, State> {
+    activitySidebarRef: React.Ref<any> = React.createRef();
+
+    detailsSidebarRef: React.Ref<any> = React.createRef();
+
+    metadataSidebarRef: React.Ref<any> = React.createRef();
+
+    versionsSidebarRef: React.Ref<any> = React.createRef();
+
+    /**
+     * Refreshes the contents of the active sidebar
+     * @returns {void}
+     */
+    refresh(): void {
+        if (this.activitySidebarRef && this.activitySidebarRef.current) {
```

Comment:
Trying to think of a readable way to reduce repetition; suggestions appreciated. I could do something like:
```
[SIDEBAR_VIEW_ACTIVITY, SIDEBAR_VIEW_DETAILS, ...].forEach((view) => {
            const key = `${view}SidebarRef`;
            if (this[key] && this[key].current) {
                this[key].current.refresh();
            }
        });
```

Reply:
We should be able to remove the identity checks, since `this.activitySidebarRef`, for example, should always be defined with the `React.createRef()` call above. One shortcut I've been using for refs is to destructure and rename them at the same time:

```
const { current: activitySidebar } = this.activitySidebar;
const { current: detailsSidebar } = this.detailsSidebar;
const { current: metadataSidebar } = this.metadataSidebar;

if (activitySidebar) {
    activitySidebar.refresh();
}

// etc...
```

You can also probably remove the `Ref` suffix on all of the property names. The flow type and use of `current` should make it clear what we're working on.